Henfam/lib/factory/menu_factory.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:Henfam/models/models.dart';
+
+//PURPOSE OF FILE IS TO MAP FIREBASE MENU DATA TO MENU MODEL
+
+class MenuFactory {
+  static Menu constructMenu(DocumentSnapshot restaurantDoc) {
+    List<Map> categoryData =
+        List<Map>.from(restaurantDoc['menu']['categories']);
+    List<MenuCategory> categories =
+        MenuFactory.constructMenuCategory(categoryData);
+    Map<String, MenuModifier> modifiers =
+        MenuFactory.constructMenuModifiers(restaurantDoc['menu']['modifiers']);
+
+    return Menu(categories: categories, modifiers: modifiers);
+  }
+
+  static List<MenuCategory> constructMenuCategory(List<Map> categoryData) {
+    List<MenuCategory> categories = [];
+
+    for (int index = 0; index < categoryData.length; index++) {
+      Map currentCategory = categoryData[index];
+      List<Map> currentItems = List<Map>.from(currentCategory['items']);
+
+      MenuCategory category = MenuCategory(
+        currentCategory['name_of_category'],
+        MenuFactory.constructMenuItems(currentItems),
+      );
+
+      categories.add(category);
+    }
+
+    return categories;
+  }
+
+  static Map<String, MenuModifier> constructMenuModifiers(Map modifierData) {
+    Map<String, MenuModifier> modifiers = {};
+
+    modifierData.forEach((key, value) {
```

Comment:
Pretty sure this handles the case where modifierData is an empty Map in the business logic side, but I'm not sure if we handle the front-end case in our other code